src/coreclr/vm/codeman.cpp

```diff
@@ -2376,6 +2376,16 @@ HeapList* LoaderCodeHeap::CreateCodeHeap(CodeHeapRequestInfo *pInfo, LoaderHeap
     }
     else
     {
+        // Include internal CodeHeap structures in the reserve
+        allocationSize = ALIGN_UP(allocationSize, VIRTUAL_ALLOC_RESERVE_GRANULARITY);
+        reserveSize = max(reserveSize, allocationSize);
+
+        if (reserveSize != (DWORD) reserveSize)
```

Comment:
This is duplicate of the condition from the top of the method, It is not particularly pretty, but it makes the fix easy to backport.

Reply:
Can you walk me through the logic of how increasing this value avoids the OOM? It is stated that at the reserve boundary this causes failures, but I don't see how increasing the amount of memory avoids that issue.

Reply:
https://github.com/dotnet/runtime/blob/a349912e4f8610f39a6ac3cb73d8d4d0968ccec4/src/coreclr/vm/codeman.cpp#L2721-L2730 is where this is called from.

When we do not find existing codeheap block that can satisfy the allocation request, we reserve address space for a new code heap block that is expected to be big enough to satisfy the allocation request, and then commit memory in the new code heap block.

This code computed the size of the new code heap block to reserve. We did not reserve enough, and the commit step failed with OOM since there was not enough reserved address space.

The problem would only show for large sizes near page size multiples. Aligning the reserved block size up to page size and above minimum would cover the problem most of the time.

Reply:
Ah, the OOM was coming from line 2730 above and not from within `LoaderCodeHeap::CreateCodeHeap`, gotcha. So we reserved space that was less than what was actually needed when we include our data structure sizes as well.